src/MongoDB.Analyzer/Core/Linq/LinqExpressionProcessor.cs

```diff
@@ -291,43 +291,10 @@ SymbolKind.Parameter or
         SimpleNameSyntax identifierNode)
     {
         var typeInfo = rewriteContext.SemanticModel.GetTypeInfo(identifierNode);
-        var remmapedType = rewriteContext.TypesProcessor.GetTypeSymbolToGeneratedTypeMapping(typeInfo.Type);
-
-        if (remmapedType == null)
-        {
-            return RewriteResult.Invalid;
-        }
-
-        SyntaxNode nodeToReplace = identifierNode;
-        var identifierName = identifierNode.Identifier.Text;
-
-        if (typeInfo.Type.TypeKind == TypeKind.Enum)
-        {
-            if (nodeToReplace.Parent is MemberAccessExpressionSyntax memberAccessExpressionSyntax)
-            {
-                nodeToReplace = memberAccessExpressionSyntax;
-                identifierName = memberAccessExpressionSyntax.Name.Identifier.Text;
-            }
-            else
-            {
-                return RewriteResult.Ignore;
-            }
-        }
-        else
-        {
-            while (nodeToReplace.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression))
-            {
-                nodeToReplace = nodeToReplace.Parent;
-            }
-        }
-
-        SyntaxNode newNode = identifierNode.Parent.Kind() switch
-        {
-            SyntaxKind.SimpleMemberAccessExpression => SyntaxFactoryUtilities.SimpleMemberAccess(remmapedType, identifierName),
-            _ => SyntaxFactory.IdentifierName(remmapedType)
-        };
+        var remappedType = rewriteContext.TypesProcessor.GetTypeSymbolToGeneratedTypeMapping(typeInfo.Type);
```

Comment:
The old code is a copy paste leftover from builders. Simple remapping is sufficient here.